tests/manage/mcg/test_write_to_bucket.py

```diff
@@ -360,3 +369,46 @@ def test_write_to_bucket_rbd_cephfs(
                     awscli_pod_session, target_dir, full_object_path, mcg_obj
                 )
             )
+
+    @tier2
+    @bugzilla("2054074")
+    @skipif_ocs_version("<4.10")
+    @pytest.mark.polarion_id("OCS-4000")
+    def test_content_encoding_with_write(self, bucket_factory, mcg_obj_session):
+        """
+        Test s3 put object operation to see if the content-encoding is stored as object
+        metadata after put
+        """
+        # create bucket
+        bucket_name = bucket_factory()[0].name
+        logger.info(f"Bucket created {bucket_name}")
+
+        # create a random file and then zip it
+        filename = f"random-{uuid4().hex}"
+        with open(filename, "wb") as f:
+            f.write(os.urandom(1000))
+        zip_obj = ZipFile(f"{filename}.zip", "w")
+        zip_obj.write(f"{filename}", compress_type=zipfile.ZIP_DEFLATED)
+        zip_obj.close()
+        logger.info(f"Random zip file generated : {filename}.zip")
+
+        # put object to the bucket created
+        s3_put_object(
+            s3_obj=mcg_obj_session,
+            bucketname=bucket_name,
+            object_key="random.zip",
+            data="random.zip",
+            content_encoding="zip",
+        )
+
+        # head object to see if the content-encoding is preserved
+        head_obj = s3_head_object(
+            s3_obj=mcg_obj_session, bucketname=bucket_name, object_key="random.zip"
+        )
+        assert (
+            head_obj["ContentEncoding"] == "zip"
+        ), "Put object operation doesn't store ContentEncoding!!"
+
+        # cleanup the files created
+        os.remove(f"{filename}.zip")
+        os.remove(f"{filename}")
```

Comment:
Should this be in teardown? If the test breaks before reaching here, the files will left uncleaned

Reply:
Good suggestion! Its better to have it that way